shared/team-building/container.tsx

```diff
@@ -89,45 +102,74 @@ const deriveUserFromUserIdFn = memoize(
     null
 )
 
-const mapStateToProps = (state, ownProps: OwnProps) => {
-  const userResults = state.chat2.teamBuildingSearchResults.getIn([
+const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
+  const teamBuildingState = state[ownProps.namespace].teamBuilding
+  const userResults = teamBuildingState.teamBuildingSearchResults.getIn([
     trim(ownProps.searchString),
     ownProps.selectedService,
   ])
 
+  const preExistingTeamMembers: I.Map<string, MemberInfo> = ownProps.teamname
+    ? state.teams.teamNameToMembers.get(ownProps.teamname)
+    : I.Map()
+
+  const disabledRoles = ownProps.teamname
+    ? getDisabledReasonsForRolePicker(state, ownProps.teamname, null)
+    : {}
```

Comment:
should hoist this so its not new every time

Reply:
hoist the empty obj?